src/test.ts

```diff
@@ -0,0 +1,176 @@
+import os from 'os';
+import S3 from './S3';
+import axios from 'axios';
+import Redis from './Redis';
+import urlParser from 'url';
+import logger from './Logger';
+import { config } from './config';
+import MediaWiki from './MediaWiki';
+import Downloader from './Downloader';
+import fs, { readFileSync } from 'fs';
+import * as QueryStringParser from 'querystring';
+import { isValidEmail, ensureTrailingChar } from './util';
+
+function IsURL(url: string) {
+  const strRegex =
+    '^((https|http|ftp|rtsp|mms)?://)' +
+    '?(([0-9a-z_!~*\'().&=+$%-]+: )?[0-9a-z_!~*\'().&=+$%-]+@)?' +
+    '(([0-9]{1,3}.){3}[0-9]{1,3}' +
+    '|' +
+    '([0-9a-z_!~*\'()-]+.)*' +
+    '([0-9a-z][0-9a-z-]{0,61})?[0-9a-z].' +
+    '[a-z]{2,6})' +
+    '(:[0-9]{1,4})?' +
+    '((/?)|' +
+    '(/[0-9a-z_!~*\'().;?:@&=+$,%#-]+)+/?)$';
+  const re = new RegExp(strRegex);
+  return re.test(url);
+}
+
+export async function test_all(argv: any) {
+  // extracting all arguments
+  const {
+    speed: _speed,
+    adminEmail,
+    mwUrl,
+    mwWikiPath,
+    mwApiPath,
+    mwRestApiPath,
+    mwModulePath,
+    mwDomain,
+    mwUsername,
+    mwPassword,
+    requestTimeout,
+    format,
+    filenamePrefix,
+    publisher: _publisher,
+    outputDirectory: _outputDirectory,
+    addNamespaces: _addNamespaces,
+    articleList: _articleList,
+    customZimFavicon,
+    optimisationCacheUrl,
+    noLocalParserFallback,
+    forceLocalParser,
+    customFlavour: customProcessorPath,
+  } = argv;
+
+  const cpuCount = os.cpus().length;
+
+  // checks for speed
+  test_speed(_speed);
+  const speed = Math.max(1, Math.round(cpuCount * (_speed || 1) * 3));
```

Comment:
This line should not be there I believe.

Reply:
@kelson42 I needed speed to declare downloader class

Reply:
That might be the problem. You should not instanciate the Downloader... it seems you just need it to download the favicon. Why not using Axios directly?